csrc/scheduler/matmul.cpp

```diff
@@ -818,8 +811,14 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
     cache_op = CacheOp::Global;
   }
 
-  acw_smem = ar->cacheAfter(load_op, cache_op);
-  bcw_smem = br->cacheAfter(load_op, cache_op);
```

Comment:
Just so that I understand correctly: previously we formed `ar` and `acw_smem`, but `ar` was just `a` unless `params.async_gmem_load_operands`. Now, we are always caching `a` directly to `acw_smem`, so there's never an intermediate register. Is that right?

Reply:
Yes!